packages/fxa-content-server/server/lib/validation.js

```diff
@@ -24,6 +24,7 @@ const PATTERNS = {
   FORM_TYPE: /^(email|other)$/,
   MIGRATION: /^(sync11|amo|none)$/,
   SERVICE: /^(sync|content-server|none|[0-9a-f]{16})$/,
+  SYNC_ENGINE: /^[\w\s.:-]+$/,
```

Comment:
Can there be whitespace in the sync engine name?

Reply:
I was erring on the side of caution because of the `.text` thing above. But I can get rid now it's using ids.